src/app/authentication/reset-password/reset-password.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { SignupService } from '../../shared/services/user/signup.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { PopupService } from '../../shared/services/popup.service';
+import { MustMatch } from '../sign-up/confirm-password-validator';
+import { SpinnerUtil } from '../../utilities/spinner-utilities';
+import { displayBackendErrorRx } from '../../utilities/error-handler';
+import { finalize, switchMap } from 'rxjs/operators';
+
+@Component({
+    selector: 'app-reset-password',
+    templateUrl: './reset-password.component.html',
+    styleUrls: ['./reset-password.component.scss']
+})
+export class ResetPasswordComponent implements OnInit {
+    resetPasswordForm: FormGroup;
+    token: string;
+
+    constructor(private formBuilder: FormBuilder,
+                private signUpService: SignupService,
+                private router: Router,
+                private popupService: PopupService,
+                private route: ActivatedRoute) {
+
+        this.resetPasswordForm = this.formBuilder.group({
+            password: new FormControl('', [Validators.required, Validators.minLength(8)]),
+            confirmPassword: new FormControl('', [Validators.required, Validators.minLength(8)]),
+        }, {
+            validator: MustMatch('password', 'confirmPassword')
+        });
+    }
+
+    ngOnInit(): void {
+        this.route.queryParams.subscribe(params => {
+            this.token = params['token'];
+        });
```

Comment:
`this.route.queryParams` is hot observable, meaning that, if you subscribe to it, and never unsubscribe, it will cause a memory leak. In this case, you don't need to change the token on queryParams change, you need it only the first time you navigate to this page. For this, you can use route snapshot and take queryParams from there as a variable, not from the subscription. Check it [here](https://angular.io/api/router/ActivatedRoute#description).